proposals/secrets.md

```diff
@@ -0,0 +1,444 @@
+# Adding Secrets and Ansible Vault Support 
+
+## Objectives
+
+- Support secrets in the Docker, K8s, and OpenShift engines
+- Use Ansible Vault as the source for secret data
+- Avoid doing the following:
+    - storing secrets in unencrypted files
+    - exposing secrets within container.yml
+    - exposing secrets in generated playbooks
+- Only decrypt vault files within the Conductor container
+- Perform decryption by handing off to `ansible-playbook`, so that decryption occurs only within the context of a playbook run.
+
+## Using a vault file
+
+A vault file will only be decrypted within the Conductor container, and the decryption will be handled by Ansible. Specifically, decryption will occur by passing vault files and credentials to `ansible-playbook`.
+ 
+The decrypted contents of a vault will only be available at playbook runtime. They will not be available within ``container.yml``.
+
+Consider the following vault file:
+
+```yaml
+---
+db_username: mysql
+db_password: openSesame!
+private_key: |
+  -----BEGIN PRIVATE KEY-----
+  MIIEvAIBADANBgkqhkiG9w0BAQEFAASCBKYwggSiAgEAAoIBAQD5koXgI24E360f
+  nhxCfOPVORzFW1CN7u/zOQdvKoIStogF0UQifDCnY/POEjoBmzBrg/UyAmsqLIli
+  xMtRIuvEhwaGEUQPoZNCaRW+1XtJ3kDvr9MVTlJTcNGOlGe/E+HyAKBq5vinxzzM
+  9ba8M9Nc1PQ93B1OTUY1QGHVYRvSFYDJ5Fnz23xKeNsnY3hmRkV7CDZXSdy9nbmy
+  1X9uz7z5bG7PKUVD3JZjI75CHAEDJKtscBv9ez/z16YTxwahIL3CXfqBq8peyAZ0
+  n4Mzj4Lt8Cwaw2Kw3w3gMhbhf4fy284+hYqHe9uqYJC6dJJSKDIXqoLSD+e8aN+v
+  BAEQcAWXAgMBAAECggEAbmHJ6HqDHJC5h3Rs11NZiWL7QKbEmCIH6rFcgmRwp0oo
+  GzqVQhNfiYmBubECCtfSsJrqhbXgJAUStqaHrlkdogx+bCmSyr8R3JuRzJerMd6l
+  Jd3EJ...
+  -----END PRIVATE KEY-----
+```
+
+Because a vault cannot be decrypted directly by Ansible Container, Referencing variable names via template strings within `container.yml` will not be supported. For example, the following `container.yml` would result in an error:
+
+```yaml
+version: '2'
+services:
+  db:
+    environment:
+      MYSQL_USERNAME: '{{ db_username }}'
+      MYSQL_PASSWORD: '{{ db_password }}'
+    command: ['/start-db.sh']
+```
+
+The only reference that will be supported is in the top-level `secrets` directive, where the secrets are represented as `key:value` pairs, and the value is variable defined in a vault. The following example will work:
+
+```yaml
+version: '2'
+services:
+  db:
+    command: ['/start-db.sh'] 
+    secrets:
+      mysql:
+        docker: 
+        - mysql_uername
+        - mysq_password
+    
+secrets:
+  # Each secret key is associated to a vault variable
+  mysql:
+    docker:
+        username: db_username   # name of variable defined in vault
+        password: db_password
+    
+    openshift:
+        username: db_username
+        password: db_password
```

Comment:
I see you did that more so down below at https://github.com/ansible/ansible-container/pull/614/files#diff-d8e1700b7b6a51785af4deeaa85232c5R378 in case you want to unify the syntax.